sale_commission/model/partner_agent.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2011 Pexego Sistemas Informáticos (<http://www.pexego.es>).
+#    All Rights Reserved
+#    $Id$
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api, _
+
+
+class res_partner_agent(models.Model):
+    """This model relates sales agents to partnes
+    """
+    _name = "res.partner.agent"
+    _rec_name = "agent_name"
+
+    partner_id = fields.Many2one(
+        "res.partner",
+        string="Partner",
+        required=True,
+        ondelete="cascade",
+        select=1
+    )
+
+    agent_id = fields.Many2one(
+        "sale.agent",
+        string="Agent",
+        required=True,
+        ondelete="cascade"
+    )
+
+    agent_name = fields.Char(
+        string="Agent name",
+        related="agent_id.name"
+    )
+
+    commission_id = fields.Many2one(
+        "commission",
+        string="Applied commission",
+        required=True,
+        ondelete="cascade"
+    )
+
+    type = fields.Selection(
```

Comment:
this field should be rename `agent_type` as `type` is a reserved name

You can use attribute `oldname` to ease field renaming.